florist/api/servers/common.py

```diff
@@ -0,0 +1,106 @@
+
+    @classmethod
+    def parse(cls, clients_info: str) -> List["ClientInfo"]:
+        """
+        Parse the client information JSON string into a ClientInfo instance.
+
+        :param clients_info: (str) A JSON string containing the client information.
+            Should be in the following format:
+            [
+                {
+                    "client": <client name as defined in florist.api.clients.common.Client>,
+                    "client_address": <Florist's client hostname and port, e.g. localhost:8081>,
+                    "data_path": <path where the data is located in the FL client's machine>,
+                    "redis_host": <hostname of the Redis instance the FL client will be reporting to>,
+                    "redis_port": <port of the Redis instance the FL client will be reporting to>,
+                }
+            ]
+        :return: (ClientInfo) an instance of ClientInfo containing the information given.
+        :raises ClientInfoParseError: If any of the required information is missing or has the
+            wrong type.
+        """
+        client_info_list: List[ClientInfo] = []
+
+        json_clients_info = json.loads(clients_info)
+        for client_info in json_clients_info:
+            if "client" not in client_info or not isinstance(client_info["client"], str):
+                raise ClientInfoParseError("clients_info does not contain key 'client'")
+            if client_info["client"] not in Client.list():
+                error_msg = f"Client '{client_info['client']}' not supported. Supported clients: {Client.list()}"
+                raise ClientInfoParseError(error_msg)
+            client = Client[client_info["client"]]
+
+            if "client_address" not in client_info or not isinstance(client_info["client_address"], str):
+                raise ClientInfoParseError("clients_info does not contain key 'client_address'")
+            client_address = client_info["client_address"]
+
+            if "data_path" not in client_info or not isinstance(client_info["data_path"], str):
+                raise ClientInfoParseError("clients_info does not contain key 'data_path'")
+            data_path = client_info["data_path"]
+
+            if "redis_host" not in client_info or not isinstance(client_info["redis_host"], str):
+                raise ClientInfoParseError("clients_info does not contain key 'redis_host'")
+            redis_host = client_info["redis_host"]
+
+            if "redis_port" not in client_info or not isinstance(client_info["redis_port"], str):
+                raise ClientInfoParseError("clients_info does not contain key 'redis_port'")
+            redis_port = client_info["redis_port"]
+
+            client_info_list.append(ClientInfo(client, client_address, data_path, redis_host, redis_port))
+
+        return client_info_list
+
+
+class ClientInfoParseError(Exception):
+    """Defines errors in parsing client info."""
+
+    pass
+
+
+class Model(Enum):
```

Comment:
I like the idea of using enumerations to set the allowable models we support. I am wondering in the future if you think we can also create enumerations for defining datasets and even losses and optimizers as well. The model part is only relevant for server but the others will allow us to have more flexibility on the client side. It will be a little tricky for model and dataset but we can perhaps separate the model type (ie MnistNet) and keyword args (ie output 1 dimension) used to instantiate it in two different arguments.